libfreerdp/core/nla.c

```diff
@@ -1210,12 +1313,78 @@ static BOOL nla_read_ts_credentials(rdpNla* nla, SecBuffer* data)
 			/* domainHint [3] OCTET STRING OPTIONAL */
 			return set_creds_octetstring_to_settings(&dec, 3, TRUE, FreeRDP_Domain, settings);
 		}
+		case TSCREDS_REMOTEGUARD:
+		{
+			/* TSRemoteGuardCreds */
+			if (!WinPrAsn1DecReadSequence(&dec, &dec2))
+				return FALSE;
+
+			/* logonCred[0] TSRemoteGuardPackageCred */
+			BOOL error = FALSE;
+			KERB_TICKET_LOGON kerbLogon;
+			WinPrAsn1Decoder logonCredsSeq = { 0 };
+			if (!WinPrAsn1DecReadContextualSequence(&dec2, 0, &error, &logonCredsSeq) || error)
+				return FALSE;
+
+			RemoteGuardPackageCredType logonCredsType;
+			wStream logonPayload;
```

Comment:
not initialized

Reply:
not really needed, if we fail to parse we exit immediatly